src/sunmatrix/cusparse/sunmatrix_cusparse.cu

```diff
@@ -1133,15 +1147,15 @@ int SUNMatMatvec_cuSparse(SUNMatrix A, N_Vector x, N_Vector y)
   {
     SUNDIALS_DEBUG_PRINT(
       "ERROR in SUNMatMatvec_cuSparse: sparse type not recognized\n");
-    return (SUNMAT_ILL_INPUT);
+    return (SUN_ERR_ARG_INCOMPATIBLE);
   }
 
 #ifdef SUNDIALS_DEBUG_CUDA_LASTERROR
   cudaDeviceSynchronize();
-  if (!SUNDIALS_CUDA_VERIFY(cudaGetLastError())) return (SUNMAT_OPERATION_FAIL);
+  if (!SUNDIALS_CUDA_VERIFY(cudaGetLastError())) return (SUN_ERR_OP_FAIL);
 #endif
 
-  return (SUNMAT_SUCCESS);
+  return SUN_SUCCESS;
 }
```

Comment:
Use `SUNAssert` in empty constructor below and handle return from `cusparseCreateCsr`